ios/RNSSearchBar.mm

```diff
@@ -163,26 +252,75 @@ - (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar
   [self resignFirstResponder];
   [self hideCancelButton];
 
-  if (self.onCancelButtonPress) {
-    self.onCancelButtonPress(@{});
+  [self emitOnCancelButtonPressEvent];
+  [self emitOnChangeTextEventWithText:_controller.searchBar.text];
+}
+#endif // !TARGET_OS_TV
+
+#pragma mark-- Fabric specific
+
+#ifdef RN_FABRIC_ENABLED
+- (void)updateProps:(facebook::react::Props::Shared const &)props
+           oldProps:(facebook::react::Props::Shared const &)oldProps
+{
+  const auto &oldScreenProps = *std::static_pointer_cast<const facebook::react::RNSSearchBarProps>(_props);
+  const auto &newScreenProps = *std::static_pointer_cast<const facebook::react::RNSSearchBarProps>(props);
+
+  [self setHideWhenScrolling:newScreenProps.hideWhenScrolling];
+
+  if (oldScreenProps.cancelButtonText != newScreenProps.cancelButtonText) {
+    [self setCancelButtonText:RCTNSStringFromStringNilIfEmpty(newScreenProps.cancelButtonText)];
   }
-  if (self.onChangeText) {
-    self.onChangeText(@{
-      @"text" : _controller.searchBar.text,
-    });
+
+  if (oldScreenProps.obscureBackground != newScreenProps.obscureBackground) {
+    [self setObscureBackground:newScreenProps.obscureBackground];
+  }
+
+  if (oldScreenProps.hideNavigationBar != newScreenProps.hideNavigationBar) {
+    [self setHideNavigationBar:newScreenProps.hideNavigationBar];
+  }
+
+  if (oldScreenProps.placeholder != newScreenProps.placeholder) {
+    [self setPlaceholder:RCTNSStringFromStringNilIfEmpty(newScreenProps.placeholder)];
+  }
+
+  if (oldScreenProps.autoCapitalize != newScreenProps.autoCapitalize) {
+    [self setAutoCapitalize:[RNSConvert UITextAutocapitalizationTypeFromCppEquivalent:newScreenProps.autoCapitalize]];
   }
+
+  // TODO: implement props with Color types
+
+  [super updateProps:props oldProps:oldProps];
+}
+
++ (facebook::react::ComponentDescriptorProvider)componentDescriptorProvider
+{
+  return facebook::react::concreteComponentDescriptorProvider<facebook::react::RNSSearchBarComponentDescriptor>();
 }
+
+#else
 #endif
 
 @end
 
+#ifdef RN_FABRIC_ENABLED
+Class<RCTComponentViewProtocol> RNSSearchBarCls(void)
+{
+  return RNSSearchBar.class;
+}
+#endif
+
 @implementation RNSSearchBarManager
 
 RCT_EXPORT_MODULE()
 
 - (UIView *)view
 {
+#ifdef RN_FABRIC_ENABLED
```

Comment:
Do we need this code here? Here we just `ifdef` the whole method: https://github.com/software-mansion/react-native-screens/blob/ba6b06bfca0537cd2183cc49e75555e93da97c1b/ios/RNSScreenStackHeaderSubview.mm#L105

Reply:
Yep, as I stated in comment of [5b8be70](https://github.com/software-mansion/react-native-screens/pull/1470/commits/5b8be70954bc25bc838d6d146b2d927c5dc0e30b), and here I changed it: [0247ea1](https://github.com/software-mansion/react-native-screens/pull/1470/commits/0247ea18fc35c098d04e71c1445837520c1e49a9)